### PR TITLE
Support SharePoint soft navigation

### PR DESCRIPTION
**Pull request by d-martins**
See [#2059](https://github.com/microsoft-search/pnp-modern-search/issues/2059)

**To test the solution:**
- Create 2 pages in a SharePoint site collection
- Add the PnP -Search Verticals web part to each
- In one page, configure 2 verticals, one for each page
  - The verticals should be hyperlinks
  - The hyperlink should be configured to open on the current tab
- Copy the settings from the web part and apply them on the second page
- Publish both pages
- Click on the tabs. Only the content area should refresh

### CODE REVIEW

File: search-parts/src/webparts/searchVerticals/components/SearchVerticalsContainer.tsx [hunk 1]
```diff
@@ -95,7 +95,17 @@ export default class SearchVerticalsContainer extends React.Component<ISearchVer
           const behavior = vertical.openBehavior === PageOpenBehavior.NewTab ? '_blank' : '_self';
           this.props.tokenService.resolveTokens(vertical.linkUrl).then((resolvedUrl: string) => {           
             resolvedUrl = resolvedUrl.replace(/\{searchTerms\}|\{SearchBoxQuery\}/gi, GlobalSettings.getValue(BuiltinTokenNames.inputQueryText));
-            window.open(resolvedUrl, behavior);
+
+            if(behavior === "_blank"){
```
Comment (wobba) [inline search-parts/src/webparts/searchVerticals/components/SearchVerticalsContainer.tsx:99]: I suggest to remove L95 and do the check on `vertical.openBehavior === PageOpenBehavior.NewTab` here

File: search-parts/src/webparts/searchVerticals/components/SearchVerticalsContainer.tsx [hunk 2]
```diff
@@ -95,7 +95,17 @@ export default class SearchVerticalsContainer extends React.Component<ISearchVer
...
+              window.open(resolvedUrl, behavior);
```
Comment (wobba) [inline search-parts/src/webparts/searchVerticals/components/SearchVerticalsContainer.tsx:100]: Use `_blank` here

File: search-parts/src/webparts/searchVerticals/components/SearchVerticalsContainer.tsx [hunk 3]
```diff
@@ -95,7 +95,17 @@ export default class SearchVerticalsContainer extends React.Component<ISearchVer
...
+            }else{
+              // Allow SharePoint to intercept the click and do a soft navigation
+              document.body.insertAdjacentHTML('beforeend', `<a target="${behavior}" href="${resolvedUrl}" style="display:none;"></a>`);
```
Comment (wobba) [inline search-parts/src/webparts/searchVerticals/components/SearchVerticalsContainer.tsx:103]: And you can remove the target attribute here.

File: search-parts/src/webparts/searchVerticals/components/SearchVerticalsContainer.tsx [hunk 4]
```diff
@@ -92,10 +92,19 @@ export default class SearchVerticalsContainer extends React.Component<ISearchVer
...
-            window.open(resolvedUrl, behavior);
+
+            if(vertical.openBehavior === PageOpenBehavior.NewTab){
+              window.open(resolvedUrl, "_blank");
+            }else{
+              // Allow SharePoint to intercept the click and do a soft navigation
+              document.body.insertAdjacentHTML('beforeend', `<a href="${resolvedUrl}" style="display:none;"></a>`);
+              const anchor = document.body.lastElementChild as HTMLElement; 
+              anchor.click();
+              document.body.removeChild(anchor);
```
Comment (wobba) [inline search-parts/src/webparts/searchVerticals/components/SearchVerticalsContainer.tsx:105]: Any chance of getting an error here if the navigation happens before the DOM is removed? Adding a try/catch around removeChild ensures that this case never happens.

Comment (d-martins) [reply search-parts/src/webparts/searchVerticals/components/SearchVerticalsContainer.tsx:105]: This is unlikely to happen, as the code block should run synchronously before the click event is resolved.
But in any case, we can validate if the anchor.parentElement is not null before removing, which would be a valid operation even if it weren't in the DOM somehow, and wouldn't result in an error.

```suggestion
              anchor.parentElement?.removeChild(anchor);
```

**Comment by wobba** [comment]
Thank you :)

**Pull request merged**